content/en/docs/tasks/manage-kubernetes-objects/kustomization.md

```diff
@@ -113,25 +105,28 @@ EOF
 The generated ConfigMap can be examined with the following command:
 
 ```shell
-BAZ=Qux kubectl kustomize ./
+kubectl kustomize ./
 ```
 
 The generated ConfigMap is:
 
 ```yaml
 apiVersion: v1
 data:
-  BAZ: Qux
   FOO: Bar
 kind: ConfigMap
 metadata:
-  name: example-configmap-1-892ghb99c8
+  name: example-configmap-1-42cfbf598f
 ```
 
 {{< note >}}
 Each variable in the `.env` file becomes a separate key in the ConfigMap that you generate. This is different from the previous example which embeds a file named `.properties` (and all its entries) as the value for a single key.
 {{< /note >}}
 
+{{< caution >}}
+Versions 4.x and older of Kustomize have a bug that loads values from the environment when they are omitted from an env file. This behaviour should not be relied on and will be removed in version 5 of Kustomize.
```

Comment:
We should document what Kubernetes (and hence `kubectl`) release this corresponds to.

Reply:
That's kinda impossible to say, since we don't have a specific timeline for v5 yet and it will depend on what upstream release is in progress when we do cut it. Should I be more vague instead, i.e. "a future release of Kustomize"?

Reply:
We [avoid statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future), unless we can be specific and can commit to the details we're providing.

How about:
> The Kustomize tool makes special use of `.env` files that is different from how most tooling treats this
> file.
> Typically, a `.env` file provides default settings for _environment variables_ that you can override, by
> setting the actual environment variable. However, Kustomize has a different interpretation. With
> Kustomize (and `kubectl kustomize`), a `.env` file specifies a set of inputs that you **should not**
> attempt to override by setting the same value in the process environment.
>
> However, there is a bug in `kubectl kustomize` that occurs when you omit an expected value from an env file.
> If you have a `.env` file that does not specify a value, and you do set the environment variable of the
> same name, then the `kubectl` tool loads that missing value. The authors of Kustomize see this behavior
> as a bug.

Reply:
How about, instead, rewriting the page so that the input filename is **not** `.env` (because this filename is associated with a convention that is the opposite of the behavior that Kustomize wants to recommend).

It's unfortunate that the field listing those filenames is `envs` (and not, eg, `parameterPaths` or `inputs`). Perhaps a different change to Kustomize or `kubectl` would also switch to a new preferred field and deprecate `envs`?